Tests/Integration/Components/ConnectExportTest.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * (c) shopware AG <info@shopware.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ShopwarePlugins\Connect\Tests\Integration\Components;
+
+use Shopware\Components\Model\ModelManager;
+use ShopwarePlugins\Connect\Components\ConfigFactory;
+use ShopwarePlugins\Connect\Components\ConnectExport;
+use ShopwarePlugins\Connect\Components\ErrorHandler;
+use ShopwarePlugins\Connect\Components\Validator\ProductAttributesValidator\ProductsAttributesValidator;
+use ShopwarePlugins\Connect\Tests\DatabaseTestCaseTrait;
+use Shopware\Models\Category\Category;
+use Tests\ShopwarePlugins\Connect\ConnectTestHelper;
+
+class ConnectExportTest extends ConnectTestHelper
+{
+    use DatabaseTestCaseTrait;
+
+    /**
+     * @var ModelManager
+     */
+    private $manager;
+
+    /**
+     * @var ConnectExport
+     */
+    private $connectExport;
+
+    public function setUp()
+    {
+        $this->manager = Shopware()->Models();
+        $this->connectExport = new ConnectExport(
+            $this->getHelper(),
+            $this->getSDK(),
+            $this->manager,
+            new ProductsAttributesValidator(),
+            ConfigFactory::getConfigInstance(),
+            new ErrorHandler(),
+            Shopware()->Container()->get('events')
+        );
+    }
+
+    public function testMarkProductsInToBeDeletedCategories()
+    {
+        $this->manager->getConnection()->executeQuery('DELETE FROM `s_articles_categories`');
+        $this->manager->getConnection()->executeQuery('DELETE FROM `s_plugin_connect_items`');
+
+        $this->manager->getConnection()->executeQuery('INSERT INTO `s_categories` (`parent`, `path`, `description`) VALUES (3, "|3|", "Test123")');
+        $categoryId = $this->manager->getConnection()->lastInsertId();
+
+        $this->manager->getConnection()->executeQuery('INSERT INTO `s_plugin_connect_items` (`article_id`, `exported`, `cron_update`) VALUES (3, 1, NULL)');
+        $this->manager->getConnection()->executeQuery('INSERT INTO `s_plugin_connect_items` (`article_id`, `exported`, `cron_update`) VALUES (4, 1, NULL)');
+
+        $this->manager->getConnection()->executeQuery('INSERT INTO `s_articles_categories` (`articleID`, `categoryID`) VALUES (?, ?)',
+            [3, $categoryId]
+        );
+        $this->manager->getConnection()->executeQuery('INSERT INTO `s_articles_categories` (`articleID`, `categoryID`) VALUES (?, ?)',
+            [4, $categoryId]
+        );
+
+        $category = new Category();
+        $category->setId($categoryId);
+
+        $this->connectExport->markProductsInToBeDeletedCategories($category);
+
+        $count = $this->manager->getConnection()->executeQuery('SELECT COUNT(*) FROM `s_plugin_connect_items` WHERE `cron_update` = 1')->fetchColumn();
+        $this->assertEquals(2, (int) $count);
+    }
+
+    public function testMarkProductsInToBeDeletedCategoriesWithChildCategories()
```

Comment:
why we need 2 different tests for this ? I think testMarkProductsInToBeDeletedCategoriesWithChildCategories also covers testMarkProductsInToBeDeletedCategories. Also you can use fixtures instead of many sql lines in your test.
you can find example in **SwagConnect/Tests/Integration/Components/ProductStream/ProductStreamServiceTest.php** 
```$this->importFixtures(__DIR__ . '/_fixtures/products_and_streams_relation.sql');```

If we decide to handle category deletion in the way that I suggested, some of the tests will be superfluous and new cases must be handled.

Reply:
good point because fixtures can be reused in lifecycle-subscriber test